lib/vast/Conversion/Core/ToLLVM.cpp

```diff
@@ -104,7 +108,12 @@ namespace vast
 
                 cond_br_lhs(rewriter, op.getLoc(), cmp_lhs, rhs_block, end_block);
 
-                rewriter.setInsertionPointToEnd(rhs_block);
+                // In the case that `rhs` consists of multiple blocks we need to
+                // insert to the last block…
+                // e.g.: in a && (b || c) mlir first matches the `||` which creates
+                // it's own cf blocks and then matches the `&&`
+                auto rhs_end_it = std::prev(end_block->getIterator(), 1);
+                rewriter.setInsertionPointToEnd(&*rhs_end_it);
```

Comment:
does this match [getSinglePredecessor](https://mlir.llvm.org/doxygen/classmlir_1_1Block.html#a72289303038da4c089aeb47ca97c29ac)?